libsolidity/formal/CHC.cpp

```diff
@@ -967,6 +986,25 @@ void CHC::connectBlocks(smtutil::Expression const& _from, smtutil::Expression co
 	addRule(edge, _from.name + "_to_" + _to.name);
 }
 
+smtutil::Expression CHC::initialConstraints(ContractDefinition const& _contract, FunctionDefinition const* _function)
+{
+	smtutil::Expression conj = state().state() == state().state(0);
+	conj = conj && errorFlag().currentValue() == 0;
+	for (auto var: stateVariablesIncludingInheritedAndPrivate(_contract))
+		conj = conj && m_context.variable(*var)->valueAtIndex(0) == currentValue(*var);
+
+	FunctionDefinition const* function = _function ?
+		_function :
+		_contract.constructor() ?
+			_contract.constructor() :
+			nullptr;
```

Comment:
I think it is enough to just set it to `_contract.constructor()` if `_function` is `nullptr`
```suggestion
		_contract.constructor();
```